### PR TITLE
Project chatbot

### PR DESCRIPTION
**Pull request by rawisou**
# Project Chatbot

This is our first paired-programming project.
Sara and I decided to build a snow bot that helps potential students pick the right ski course using
basic javaScript.

**Users should be able to :**
- Enter name and age
- For age input, when users input something that is not a number, they will be asked to re-enter their info using numbers
- If users are younger than 25 years old, the chat will end and the page will refresh.
- If users are 26 years old or older, the chat will proceed to the next question.

## View it live

https://distracted-stonebraker-d0c8bd.netlify.app/

### CODE REVIEW

File: code/script.js [hunk 1]
```diff
@@ -32,16 +37,177 @@ const showMessage = (message, sender) => {
...
-// normally we would invoke a function like this:
-// greeting()
-// But if we want to add a little delay to it, we can wrap it in a setTimeout:
-// setTimeout(functionName, timeToWaitInMilliSeconds)
 // This means the greeting function will be called one second after the website is loaded.
-setTimeout(greeting, 1000)
+setTimeout(greeting, 900)
+
+const reply = (input) => {
+  showMessage(input, 'user')
```
Comment (JaEngd) [inline code/script.js:47]: Add a function, for example; userInput.value = '', to clear name input to an empty string.

File: code/script.js [hunk 2]
```diff
@@ -32,16 +37,177 @@ const showMessage = (message, sender) => {
```
Comment (JaEngd) [inline code/script.js:37]: When there are many text-boxes, the buttons overlaps the text a bit. Add a function that makes the chat scroll to the end of the last message, for example; chat.scrollTop = chat.scrollHeight.

File: code/style.css [hunk 3]
```diff
@@ -138,3 +138,16 @@ button:hover {
   opacity: 0.9;
   transition: all 0.2s ease;
 }
+
```
Comment (JaEngd) [inline code/style.css:141]: Text appears outside the text box at one stage. Fix it by set a media query in mobile view or by adding flex-shrink in css for example.

File: code/script.js [hunk 4]
```diff
@@ -32,16 +37,177 @@ const showMessage = (message, sender) => {
...
+
+
+let flag = 'name'
+sendButton.addEventListener('click', () => {
+  if (flag === 'name') {
+    reply(userInput.value)
+    showMessage(`Thank you for your interest, ${userInput.value}. May I know your age?`, 'bot')
+    flag = 'age'
+  }
+  else if (flag = 'age') {
```
Comment (JaEngd) [inline code/script.js:66]: Great job with the error-handling part using the if/else statement.

File: code/script.js [hunk 5]
```diff
@@ -32,16 +37,177 @@ const showMessage = (message, sender) => {
...
+        return false;
+      } else {
+        return setTimeout(skiStyle, 900)
+      }
+    }
+  }
+})
+
+function isNumeric(val) {
+  return /^-?\d+$/.test(val);
```
Comment (JaEngd) [inline code/script.js:87]: Interesting, what does this do (return /^-?\d+$/.test(val);) ?

**Review by JaEngd — APPROVED** [review]
Great work! You've met the requirements, the code is well structured and you've done a good job creating different options and scenarios for the user.

Good job with the if/else statement. I like the fact that you've also added some error-handling.

I've made some comments about the responsiveness in mobile view and some of the chat-bot functions.

**Review by JaEngd — APPROVED** [review]
Test

File: code/script.js (hunk 5 above)
Comment (rawisou) [reply code/script.js:87]: Sorry I just saw this now! This is Regular Expression. I used this to check if what the user typed was number or not. I find [this website](https://bytes.com/topic/javascript/answers/150159-what-does-mean-d-test-el-value) very helpful if you want to read more about this. :)

**Pull request closed without merging**